pages/product/[productId]/index.tsx

```diff
@@ -0,0 +1,47 @@
+import { Panel, Text } from "@bigcommerce/big-design";
+import { useRouter } from "next/router";
+import ErrorMessage from "@components/error";
+import Loading from "@components/loading";
+import { useProductInfo, useProductList } from "@lib/hooks";
+
+const SidePanel = () => {
+  const router = useRouter();
+  const productId = Number(router.query?.productId);
+  const { error, isLoading, list = [] } = useProductList();
+  const { isLoading: isInfoLoading, product } = useProductInfo(productId, list);
```

Comment:
🍹 Seems like we could make `list` an optional parameter for `useProductInfo` or even just pass in an empty array so that we just fetch the one product instead of the whole list and then filtering it. Then we can get rid of `useProductList` and only have to check one `isLoading` condition. Might be for another ticket, just noticed this pattern and making a note.

Reply:
Ohh yeah that could be a could idea, will keep this in mind and bring it up for a future change to improve this addition!